SchoolApplication/School.Repository/StudentRepository.cs

```diff
@@ -14,123 +14,177 @@ public class StudentRepository : BaseRepository<Student>
     {
         #region student Repository
         public List<StudentDtoPost> GetStudents()
-        {// BaseRepository<Student> b = new BaseRepository<Student>();
-         //  Add(obj);
+        {
             List<StudentDtoPost> DtoList = new List<StudentDtoPost>();
             try
             {
                 using (SchoolContext db = new SchoolContext())
+                {            //Simple Join to gett Student With count of Courses
+                    var students = (from s in db.Students
+                                    select new StudentDtoPost()
+                                    {
+                                        student = s,
+                                        CoursesCount = (from sc in db.StudentCourses.Where(x => x.StudentId == s.Id)
+                                                        select sc.StudentId).Count()
+                                    }).ToList();
+                    return students;
+                    //used before to get Coursecount without Jooin
+                    //var StdList = db.Students.ToList();
+
+                    //foreach (var student in StdList)
```

Comment:
Why is there a commented code??? 
If you want to use it later, put comment on how and when